### ISSUE TITLE
feat(core): Add all currencies in  Currency Enum

### ISSUE THREAD

**Issue by prasunna09** [juspay/hyperswitch#1814]
### Feature Description

Add missing currencies to Currency enum in [enum.rs](https://github.com/juspay/hyperswitch/blob/5ad3950892fc0c84b26092b0732dd18d2d913d12/crates/common_enums/src/enums.rs#L291), since we want descent coverage of the currencies. Reference - https://en.wikipedia.org/wiki/List_of_circulating_currencies

### Possible Implementation

1. Add all the missing currencies.
2. Run the migrations (Can refer to this pr - https://github.com/juspay/hyperswitch/pull/1670/files#diff-b90f2b50c9b9e664a9f6b840c1590342362713cffb3546db779813106242772f)

### Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

None

**Comment by tusharxoxoxo** [comment]
@prasunna09 going to take a look at this by the evening, and will submit it in morning

**Comment by prasunna09** [comment]
Then I'll assign this to you! @tusharxoxoxo

**Comment by tusharxoxoxo** [comment]
what should be done for these currencies whose iso code is missing?
![image](https://github.com/juspay/hyperswitch/assets/79051850/e3c98038-8407-42c3-8215-28bbb31ae15c)

In this issue, do we only need to add the missing currencies in `pub enum Currency` or do we also need to implement them with their respective iso_4212 code in the very next function and also the other parts such as `pub fn is_three_decimal_currency(self) -> bool {` and `pub fn is_zero_decimal_currency` accordingly??
![image](https://github.com/juspay/hyperswitch/assets/79051850/db51f845-bcf4-40fd-ae14-cfc59b7fe506)

**Comment by prasunna09** [comment]
Yes, you have to add all the missing currencies in `pub enum currency` and have to add their corresponding `iso_4217` codes and also include them in other `is_three_decimal_currency(self) -> bool` and `pub fn is_zero_decimal_currency`.

**Comment by tusharxoxoxo** [comment]
there r some currencies which are mentioned in this Wikipedia article [List_of_circulating_currencies](https://en.wikipedia.org/wiki/List_of_circulating_currencies) which r not present in [ISO_4217](https://en.wikipedia.org/wiki/ISO_4217) like, for example, SVC is not present in the list of currencies, but is present in ISO_4217 list of currencies and vice versa CNY is present in the list of currencies but is not present in the ISO_4217 list

**Comment by tusharxoxoxo** [comment]
`HRK` is also not present in ISO_4217 list

**Comment by tusharxoxoxo** [comment]
> There r some currencies which are mentioned in this Wikipedia article [List_of_circulating_currencies](https://en.wikipedia.org/wiki/List_of_circulating_currencies) which r not present in [ISO_4217](https://en.wikipedia.org/wiki/ISO_4217) like, for example, `SVC` is not present in the list of currencies, but is present in ISO_4217 list of currencies and vice versa `CNY` is present in the list of currencies but is not present in the ISO_4217 list.

my bad `CNY` is present in the list

**Comment by tusharxoxoxo** [comment]
some of the currencies go to the 4 decimal digits so for them should I create a new function or leave them as it is 
`CLF`

**Comment by tusharxoxoxo** [comment]
there r a lot of currencies that don't have anything in the place decimal row like it's blank
![image](https://github.com/juspay/hyperswitch/assets/79051850/90ef3d95-8e5f-4cae-8613-5f4259925bea)
